app/Api/V1/Domain/Problem/Services/ReadProblemService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Api\V1\Domain\Problem\Services;
+
+
+use App\Api\V1\Domain\Problem\Entity\Problem;
+use App\Api\V1\Domain\Problem\Param\ReadProblemParam;
+use App\Api\V1\Domain\Problem\Repository\ProblemRepository;
+use Illuminate\Database\Eloquent\Collection;
+
+class ReadProblemService
+{
+    protected $repository;
+
+    public function __construct(ProblemRepository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function readMany(ReadProblemParam $param) : Collection
+    {
+        $param = $param->toArray();
+
+        $offset = $param[ReadProblemParam::ATTRIBUTE_PAGE]*$param[ReadProblemParam::ATTRIBUTE_LIMIT];
```

Comment:
To get parameter from Param object you should declare getter function (`getPage` and `getLimit` or `getOffset`)